classes/PSShopgatePlugin.php

```diff
@@ -645,11 +788,26 @@ protected function prepareAddresses(ShopgateOrder $order)
 	protected function getRequiredAddressFields()
 	{
 		$address = new Address();
-		return $address->getFieldsRequiredDatabase();
+
+		if (method_exists($address, 'getFieldsRequiredDatabase'))
+		{
+			/**
+			 * getFieldsRequiredDatabase
+			 */
+			return $address->getFieldsRequiredDatabase();
```

Comment:
This function exists from 1.4.0.2 to the last version of PrestaShop. You can directly return the result of this function without checking if it exists.